src/js/components/Grommet/Grommet.js

```diff
@@ -19,26 +20,28 @@ class Grommet extends Component {
   static displayName = 'Grommet';
 
   static getDerivedStateFromProps(nextProps, prevState) {
-    const { dir, theme = {}, themeMode } = nextProps;
+    const {
+      background: backgroundProp,
+      dir,
+      theme = {},
+      themeMode,
+    } = nextProps;
     const { theme: stateTheme, themeProp, themeModeProp } = prevState;
 
     const nextTheme = deepMerge(baseTheme, theme);
     if (!stateTheme || theme !== themeProp || themeMode !== themeModeProp) {
       const {
-        colors: { background },
+        colors: { background: themeBackground },
       } = nextTheme.global;
-      // Determine whether to start in dark or light mode.
-      if (typeof background === 'object') {
-        // background is an object, use themeMode, theme default
-        // or first key
-        const color =
-          background[
-            themeMode || nextTheme.defaultMode || Object.keys(background)[0]
-          ];
-        nextTheme.dark = color ? colorIsDark(color) : false;
-      } else if (nextTheme.dark === undefined) {
-        nextTheme.dark = (background && colorIsDark(background)) || false;
-      }
+
+      nextTheme.dark = (themeMode || theme.defaultMode) === 'dark';
```

Comment:
I think we would like a comment here for when we read this months from now and forget why we are setting `nextTheme.dark` twice.